Source/WebCore/page/RemoteFrame.cpp

```diff
@@ -81,8 +84,18 @@ void RemoteFrame::didFinishLoadInAnotherProcess()
 {
     m_preventsParentFromBeingComplete = false;
 
-    if (auto* ownerElement = this->ownerElement())
+    if (auto* ownerElement = this->ownerElement()) {
         ownerElement->document().checkCompleted();
+        RefPtr widgetRenderer = dynamicDowncast<RenderWidget>(ownerElement->renderer());
+        if (!widgetRenderer)
+            return;
+
+        if (!widgetRenderer->hasLayer() || !widgetRenderer->layer()->isComposited())
+            return;
+
+        if (auto frameHostingNodeID = widgetRenderer->layer()->backing()->scrollingNodeIDForRole(ScrollCoordinationRole::FrameHosting))
+            page()->scrollingCoordinator()->setLayerHostingContextIdentifier(frameHostingNodeID, m_layerHostingContextIdentifier);
```

Comment:
This probably isn't the best place to set this. What if this iframe is unparented and re-parented (needs a test)? Do we need to hook things up before the load finishes (pages can display content before the load event fires).

I think it would make more sense to call this in `RenderLayerCompositor::updateScrollingNodeForFrameHostingRole`.

Reply:
Right fixed.